sycl/source/queue.cpp

```diff
@@ -218,13 +218,22 @@ bool queue::device_has(aspect Aspect) const {
 bool queue::begin_recording(
     ext::oneapi::experimental::command_graph<
         ext::oneapi::experimental::graph_state::modifiable> &graph) {
-  // Empty Implementation
+  using namespace ext::oneapi::experimental;
+  if (!impl->getCommandGraph()) {
+    impl->setCommandGraph(
+        sycl::detail::getSyclObjImpl<command_graph<graph_state::modifiable>>(
+            graph));
+    return false;
```

Comment:
should these `return true/false` values be swapped? Think we should be returning `true` on successfully starting to record

Reply:
Ah good spot, you are correct. This is now fixed!